File: src/main/java/orbapi/Marshaller.java
package orbapi;

import vson.Vson;
import java.lang.reflect.Method;


public class Marshaller {


    public static byte[] marshallMethod(Method m, Object[] args){

        MethodCall methodCall = new MethodCall(m.getName(),args);
        return Vson.toJson(methodCall).getBytes();
    }

    public static MethodCall unMarshallMethod(byte[] req){
         return (MethodCall)Vson.fromJson(new String(req));

    }

    public static byte[] marshallObject(Object obj){
        return Vson.toJson(obj).getBytes();
    }

    public static Object unMarshallObject(byte[] reply){
        return Vson.fromJson(new String(reply));
    }
}

File: ObjectRequestBrokerCS/ORB/orbapi/Marshaller.cs
using ORB.vson;

namespace ORB.orbapi
{


	public static class Marshaller
	{

		public static byte[] MarshallObject(object obj)
		{
			return ToByteArray(Vson.ToJson(obj));
		}

		public static object UnMarshallObject(byte[] reply)
		{
			return Vson.FromJson(StringFromByteArray(reply));
		}

		private static byte[] ToByteArray(string str)
		{
			return System.Text.Encoding.ASCII.GetBytes(str);
		}

		public static string StringFromByteArray(byte[] bytes)
		{
			return System.Text.Encoding.ASCII.GetString(bytes);
		}
	}

}